### ISSUE TITLE
making gifs with magick is too slow

### ISSUE THREAD

**Issue by roblanf** [danlwarren/RWTY#164]
NO idea what's happening here yet. More details are here though:
https://discuss.ropensci.org/t/help-with-magick-package/1901

**Comment by roblanf** [comment]
so far no luck, but perhaps the simplest solution would be the best:
https://stackoverflow.com/questions/54109437/multiple-gganimate-plots-side-by-side

i.e. just put the two animations into the R markdown document like this:

```{r}
animate(anime_one)
```

```{r}
animate(anime_two)
```

this will save time and headaches, and be absolutely fine.

**Comment by roblanf** [comment]
This has been bugging me all day. So here's another idea. We dig into gganimate a bit and do something like this e.g. for the trace.animation...

basically we need to deconstruct the `animate` function so we can get at the list of png files written by gganimate. Once we have these lists for both plots, we can stitch them together with magick and then output the gif. This will stop the png->gif->png transformations we currently do, which (a) take a lot of time and (b) screw up the animation anyway.

The basic idea looks something like:

```
trace.plot = gganimate:::prerender(trace.animation, n.frames)
trace.pngs = gganimate:::draw_frames(plot, 1:n.frames, device = "png", 1)
```

that gets us the list of pngs from one plot. We can do the same for the space plot, then use something like this to stack them (needs editing, but the idea is the same

```
  final_gif <- image_append(c(space_mgif[1], trace_mgif[1]), stack = TRUE)
  for(i in 2:n.frames){
    combined <- image_append(c(space_mgif[i], trace_mgif[i]), stack = TRUE)
    final_gif <- c(final_gif, combined)
  }
```

**Comment by roblanf** [comment]
question for @danlwarren, are we allowed to use triple colons (wooo!) in a package or does CRAN get us in trouble for that?

**Comment by danlwarren** [comment]
Nope, CRAN check and Travis will both barf on triple colons.  I tried it with ENMTools a few weeks back.
